app/services/rss_notifications_manager.rb

```diff
@@ -0,0 +1,42 @@
+class RssNotificationsManager
+  def initialize(options = {})
+    @url = options[:url] || 'https://theforeman.org/feed.xml'.freeze
+    @group = options[:group] || _('Community')
+    @latest_posts = options[:latest_posts] || 3
+    @expiration_date = options[:expiration_date] || 1.week
+    @force_repost = options[:force_repost] || false
+    @audience = options[:audience] || Notification::AUDIENCE_GLOBAL
+  end
+
+  def create_notifications
+    feed = RSS::Parser.parse(@url, false)
+    feed.items[0, @latest_posts].each do |item|
+      name = item.title.content
+      summary = item.summary.content # display summary if available
+      link = item.link.href # link href is indispensable
+
+      # Find previous notification blueprints with this name.
+      if (blueprint = NotificationBlueprint.find_by_name(name))
+        # If some exist use them
+        next unless @force_repost
+      else
+        # If none exist, create one.
+        blueprint = NotificationBlueprint.new(
```

Comment:
does this really works? I would think blueprint is only defined inside the else block? or does it work as its part of if statement? regardless, I would break it down to multiple methods, so its more readable and testable.